Add prompt template retrieval endpoint

abubakarsohail opened the pull request:
This pull request adds a new endpoint for retrieving prompt templates by identifier. The endpoint is `/prompt-templates/{identifier}` and can be accessed with a POST request. This allows users to fetch prompt templates using the identifier as a parameter. The implementation includes updates to the Python SDK and JavaScript SDK to demonstrate how to use the new endpoint.

jzone3 commented:
![CleanShot 2024-01-15 at 15 07 58@2x](https://github.com/MagnivOrg/prompt-layer-docs/assets/1629612/a9584ad1-ccbc-489e-9e32-4021b819e4f7)
![CleanShot 2024-01-15 at 15 06 53@2x](https://github.com/MagnivOrg/prompt-layer-docs/assets/1629612/d1e30cd2-16db-45b3-b6d7-29f6c5b84897)
![CleanShot 2024-01-15 at 15 06 36@2x](https://github.com/MagnivOrg/prompt-layer-docs/assets/1629612/cc164445-cb65-4d77-8b93-5716cdae41bc)
![CleanShot 2024-01-15 at 15 05 34@2x](https://github.com/MagnivOrg/prompt-layer-docs/assets/1629612/98d9404f-66e3-45da-9b41-f199a6251f58)
![CleanShot 2024-01-15 at 15 05 15@2x](https://github.com/MagnivOrg/prompt-layer-docs/assets/1629612/e535eb98-6566-4ac0-b656-577a0d664bb0)

jzone3 reviewed (changes requested):
See screenshots

abubakarsohail commented:
> See screenshots

@jzone3 updated the listing. the `Get Prompt Templates` is the endpoint for fetching all prompt templates.

Updated the wording for `Note`

Removed legacy code snippets.

jzone3 commented on reference/get-prompt-template.mdx:
```diff
@@ -1,5 +1,5 @@
 ---
-title: "Get Prompt Template"
+title: "Get Prompt Template (Depcrecated)"
```
spelling mistake

jzone3 commented on features/prompt-registry.mdx:
````diff
@@ -64,14 +54,21 @@ Release labels like `prod` and `staging` can be optionally applied to template v
 
 ```python Python SDK
-promptlayer.prompts.get("my_template", label="prod")
+promptlayer.templates.get(
+  "my_template",
+  {
+    "label": "prod"
+  }
+)
````
Please fix indentation, this should be one line

jzone3 commented on features/prompt-registry.mdx:
```diff
@@ -64,14 +54,21 @@ Release labels like `prod` and `staging` can be optionally applied to template v
-  prompt_name: "my_template",
-  label: "prod"
-});
+const template_dict = await promptlayer.templates.get(
```
One line

jzone3 commented on features/prompt-registry.mdx:
````diff
@@ -83,13 +80,22 @@ You can also optionally pass `version` to get an older version of a prompt. By d
 
 ```python Python SDK
-template_dict = promptlayer.prompts.get("my_template", version=3)
+template_dict = promptlayer.templates.get(
````
One line

jzone3 commented on features/prompt-registry.mdx:
```diff
@@ -83,13 +80,22 @@ You can also optionally pass `version` to get an older version of a prompt. By d
-const template_dict = await promptlayer.prompts.get({
-  prompt_name: "my_template", version: 3
-});
+const template_dict = await promptlayer.templates.get(
```
one line

abubakarsohail replied:
One line? Do you wane me to move print statement to one line?

jzone3 commented on features/prompt-registry.mdx:
```diff
@@ -32,30 +32,20 @@ It’s simple.
 Alternatively, use the REST API endpoint `/rest/get-prompt-template` ([read more](/reference/get-prompt-template)).
 
-_Note:_ We have a new prompt template schema that is more flexible and powerful. If you are using the new schema, you can use the `promptlayer.templates.get` method to get a template. If you are using the old schema, you can use the `promptlayer.prompts.get` method to get a template. Notice how `promptlayer.prompts.get` will always return templates in legacy shape, while `promptlayer.templates.ge ...
+*Note:* We have deprecated the legacy prompt template schema. Use `promptlayer.templates.get` to retrieve prompt templates. Notice how `promptlayer.prompts.get` will always return templates in legacy shape. 
```
Note: Please use `promptlayer.templates.get` to access prompt templates. The `promptlayer.prompts.get` function is deprecated and returns templates in the legacy format.

jzone3 commented on features/prompt-registry.mdx:
```diff
@@ -32,30 +32,20 @@ It’s simple.
+*Note:* We have deprecated the legacy prompt template schema. Use `promptlayer.templates.get` to retrieve prompt templates. Notice how `promptlayer.prompts.get` will always return templates in legacy shape. 
 
-For REST API of fetching a prompt template in new schema, please refer to `/prompt-templates` ([read more](/reference/templates-get)).
+For REST API of fetching a prompt template, please refer to `/prompt-templates` ([read more](/reference/templates-get)).
```
To fetch a prompt template using the REST API, refer to the endpoint endpoint `/prompt-templates` ([read more](/reference/templates-get)).

jzone3 reviewed (changes requested):
A few comments. Doesn't look like you updated the REST API list that I asked for in the screenshot

jzone3 commented on features/prompt-registry.mdx:
```diff
@@ -487,7 +487,38 @@ The following code snippet shows you how to do this:
+    }
+  }
+)
+response = openai.chat.completions.create(
```
Why is this `openai.chat` instead of `client.chat`?

abubakarsohail replied:
updated to client
